Graphite build with VS 2019 raises warning C4996 about deprecated `std::iterator` usage

### Godot version

4.0.beta (52aae6d4f)

### System information

Windows 10 64-bit, VS 2019, MSVC 14.29.30133

### Issue description

When building Godot on Windows 10 with VS 2019 with `scons p=windows`, the following warnings are raised when building the Graphite thirdparty code as part of the `text_server_adv` module:
```
cl /Fothirdparty\graphite\src\Segment.windows.tools.x86_64.obj /c thirdparty\graphite\src\Segment.cpp /TP /w /std:c++17 /nologo /Zi /FS /Od /EHsc /MT /Gd /GR /utf-8 /bigobj /wd4267 /wd4244 /wd4305 /wd4018 /wd4800 /EHsc /w -DGRAPHITE2_STATIC -DGRAPHITE2_NTRACING -DGRAPHITE2_NFILEFACE /DDEBUG_ENABLED /DDEV_ENABLED /DNO_EDITOR_SPLASH /DWINDOWS_ENABLED /DWASAPI_ENABLED /DWINMIDI_ENABLED /DTYPED_METHOD_BIND /DWIN32 /DMSVC /DWINVER=0x0601 /D_WIN32_WINNT=0x0601 /DNOMINMAX /D_WIN64 /DVULKAN_ENABLED /DGLES3_ENABLED /DTOOLS_ENABLED /DMINIZIP_ENABLED /DZSTD_STATIC_LINKING_ONLY /DUSE_VOLK /DVK_USE_PLATFORM_WIN32_KHR /DGLAD_ENABLED /DGLES_OVER_GL /Ithirdparty\graphite\src /Ithirdparty\graphite\include /Ithirdparty\libpng /Ithirdparty\glad /Ithirdparty\volk /Ithirdparty\vulkan /Ithirdparty\vulkan\include /Ithirdparty\zstd /Ithirdparty\zlib /Iplatform\windows /I.
Position.cpp
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xutility(1169): warning C4996: 'std::iterator<std::input_iterator_tag,std::pair<graphite2::sparse::key_type,graphite2::sparse::mapped_type>,ptrdiff_t,std::pair<graphite2::sparse::key_type,graphite2::sparse::mapped_type> *,std::pair<graphite2::sparse::key_type,graphite2::sparse::mapped_type> &>::iterator_category': warning STL4015: The std::iterator class template (used as a base class to provide typedefs) is deprecated in C++17. (The <iterator> header is NOT deprecated.) The C++ Standard has never required user-defined iterators to derive from std::iterator. To fix this warning, stop deriving from std::iterator and start providing publicly accessible typedefs named iterator_category, value_type, difference_type, pointer, and reference. Note that value_type is required to be non-const, even for constant iterators. You can define _SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING or _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have received this warning.
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xutility(1189): note: see reference to class template instantiation 'std::_Iterator_traits_base<_Iter,void>' being compiled
        with
        [
            _Iter=`anonymous-namespace'::glat_iterator
        ]
C:\Users\Remi\Documents\Godot\godot.git\thirdparty\graphite\src\inc/Sparse.h(106): note: see reference to class template instantiation 'std::iterator_traits<I>' being compiled
        with
        [
            I=`anonymous-namespace'::glat_iterator
        ]
C:\Users\Remi\Documents\Godot\godot.git\thirdparty\graphite\src\inc/GlyphFace.h(75): note: see reference to function template instantiation 'graphite2::sparse::sparse<I>(I,const I)' being compiled
        with
        [
            I=`anonymous-namespace'::glat_iterator
        ]
thirdparty\graphite\src\GlyphCache.cpp(420): note: see reference to function template instantiation 'graphite2::GlyphFace::GlyphFace<`anonymous-namespace'::glat_iterator>(const graphite2::Rect &,const graphite2::Position &,I,const I)' being compiled
        with
        [
            I=`anonymous-namespace'::glat_iterator
        ]
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xutility(1170): warning C4996: 'std::iterator<std::input_iterator_tag,std::pair<graphite2::sparse::key_type,graphite2::sparse::mapped_type>,ptrdiff_t,std::pair<graphite2::sparse::key_type,graphite2::sparse::mapped_type> *,std::pair<graphite2::sparse::key_type,graphite2::sparse::mapped_type> &>::value_type': warning STL4015: The std::iterator class template (used as a base class to provide typedefs) is deprecated in C++17. (The <iterator> header is NOT deprecated.) The C++ Standard has never required user-defined iterators to derive from std::iterator. To fix this warning, stop deriving from std::iterator and start providing publicly accessible typedefs named iterator_category, value_type, difference_type, pointer, and reference. Note that value_type is required to be non-const, even for constant iterators. You can define _SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING or _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have received this warning.
Segment.cpp
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xutility(1171): warning C4996: 'std::iterator<std::input_iterator_tag,std::pair<graphite2::sparse::key_type,graphite2::sparse::mapped_type>,ptrdiff_t,std::pair<graphite2::sparse::key_type,graphite2::sparse::mapped_type> *,std::pair<graphite2::sparse::key_type,graphite2::sparse::mapped_type> &>::difference_type': warning STL4015: The std::iterator class template (used as a base class to provide typedefs) is deprecated in C++17. (The <iterator> header is NOT deprecated.) The C++ Standard has never required user-defined iterators to derive from std::iterator. To fix this warning, stop deriving from std::iterator and start providing publicly accessible typedefs named iterator_category, value_type, difference_type, pointer, and reference. Note that value_type is required to be non-const, even for constant iterators. You can define _SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING or _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have received this warning.
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xutility(1172): warning C4996: 'std::iterator<std::input_iterator_tag,std::pair<graphite2::sparse::key_type,graphite2::sparse::mapped_type>,ptrdiff_t,std::pair<graphite2::sparse::key_type,graphite2::sparse::mapped_type> *,std::pair<graphite2::sparse::key_type,graphite2::sparse::mapped_type> &>::pointer': warning STL4015: The std::iterator class template (used as a base class to provide typedefs) is deprecated in C++17. (The <iterator> header is NOT deprecated.) The C++ Standard has never required user-defined iterators to derive from std::iterator. To fix this warning, stop deriving from std::iterator and start providing publicly accessible typedefs named iterator_category, value_type, difference_type, pointer, and reference. Note that value_type is required to be non-const, even for constant iterators. You can define _SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING or _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have received this warning.
C:\Program Files (x86)\Microsoft Visual Studio\2019\Community\VC\Tools\MSVC\14.29.30133\include\xutility(1173): warning C4996: 'std::iterator<std::input_iterator_tag,std::pair<graphite2::sparse::key_type,graphite2::sparse::mapped_type>,ptrdiff_t,std::pair<graphite2::sparse::key_type,graphite2::sparse::mapped_type> *,std::pair<graphite2::sparse::key_type,graphite2::sparse::mapped_type> &>::reference': warning STL4015: The std::iterator class template (used as a base class to provide typedefs) is deprecated in C++17. (The <iterator> header is NOT deprecated.) The C++ Standard has never required user-defined iterators to derive from std::iterator. To fix this warning, stop deriving from std::iterator and start providing publicly accessible typedefs named iterator_category, value_type, difference_type, pointer, and reference. Note that value_type is required to be non-const, even for constant iterators. You can define _SILENCE_CXX17_ITERATOR_BASE_CLASS_DEPRECATION_WARNING or _SILENCE_ALL_CXX17_DEPRECATION_WARNINGS to acknowledge that you have received this warning.
```

I'm not sure why as the `text_server_adv` module properly disabled warnings (and `-w` is present on the command line).
I guess this one doesn't get silenced, or it's raised by actually building a Godot file that includes a graphite file?

We could possibly pass `_SILENCE_ALL_CXX17_DEPRECATION_WARNINGS` to the `env_graphite` env to silence it since it's not our problem to fix.

CC @bruvzg

### Steps to reproduce

- `scons p=windows` with VS 2019

### Minimal reproduction project

_No response_

--- a/modules/text_server_adv/SCsub
+++ b/modules/text_server_adv/SCsub
@@ -215,6 +215,9 @@ if env["builtin_graphite"] and freetype_enabled and env["graphite"]:
         ]
     )
 
+    if env.msvc:  # Not our business to fix.
+        env_graphite.Append(CCFLAGS=["-D_SILENCE_ALL_CXX17_DEPRECATION_WARNINGS"])
+
     lib = env_graphite.add_library("graphite_builtin", thirdparty_sources)
     thirdparty_obj += lib
 

